Docs RFEs and release notes for 1.19 (#10819)
solo-io/gloo: docs/content/operations/upgrading/faq.md

### Breaking changes

> No breaking changes are reported for this release

Reviewer: ```suggestion
No breaking changes are reported for this release.
```

When referencing a gRPC OpenTelemetry collector in your Gateway, Gloo Gateway automatically generates an Envoy configuration that sets the cluster name as the `:authority` pseudo-header. If your collector expects a different `:authority` header, you can specify that by setting the `spec.httpGateway.options.httpConnectionManagerSettings.tracing.openTelemetryConfig.grpcService.authority` value on your Gateway as shown in the following example. 

{{< highlight yaml "hl_lines=23-24" >}}
apiVersion: gateway.solo.io/v1
kind: Gateway
metadata:
  labels:
    app: gloo
    app.kubernetes.io/name: gateway-proxy-tracing-authority
  name: gateway-proxy-tracing-authority
spec:
  bindAddress: '::'
  bindPort: 18082
  proxyNames:
    - gateway-proxy
  httpGateway:
    virtualServiceSelector:
      gateway-type: tracing
    options:
      httpConnectionManagerSettings:
        tracing:
          openTelemetryConfig:
            collectorUpstreamRef:
              name: opentelemetry-collector
              namespace: default
            grpcService:
              authority: my-authority
{{< /highlight >}}
        
### Log filter state in gRPC access logs

> Enable grpc access logging of filter state with

Reviewer: Maybe this sentence needs to be deleted? or it was cut short.
```suggestion
```

          staticClusterName: access_log_cluster
          filterStateObjectsToLog:
          - io.solo.modsecurity.audit_log
```

{{% /tab %}}
{{< /tabs >}}

Example access log output: 
```
"filter_state_objects":{"io.solo.modsecurity.audit_log":{"type_url":"type.googleapis.com/google.protobuf.StringValue","value":"Ct0DLS0tQS0tClsxMC9NYXIvMjAyNToxNDoyNTowMSArMDAwMF0gMTc0MTYxNjcwMTQ1LjkwMTMzMCAxMjcuMC4wLjEgMzg5NTQgIDAKLS0tQi0tCkdFVCAvYWxsLXBldHMgSFRUUC8xLjEKOm1ldGhvZDogR0VUCmhvc3Q6IGxvY2FsaG9zdDo4MDgwCjpwYXRoOiAvYWxsLXBldHMKOmF1dGhvcml0eTogbG9jYWxob3N0OjgwODAKeC1mb3J3YXJkZWQtcHJvdG86IGh0dHAKOnNjaGVtZTogaHR0cAp1c2VyLWFnZW50OiBjdXJsLzguMTEuMAphY2NlcHQ6ICovKgp4LXJlcXVlc3QtaWQ6IDUwZTBjZTVmLTk0NjktNGQ3NS1hMzUxLWU4MmI0ODFlYTFmYgoKLS0tRi0tCkhUVFAvMS4xIDIwMAo6c3RhdHVzOiAyMDAKY29udGVudC10eXBlOiBhcHBsaWNhdGlvbi94bWwKZGF0ZTogTW9uLCAxMCBNYXIgMjAyNSAxNDoyNTowMSBHTVQKY29udGVudC1sZW5ndGg6IDg2CngtZW52b3ktdXBzdHJlYW0tc2VydmljZS10aW1lOiAyCgotLS1ILS0KCi0tLVotLQoK"}}}
```

For more information, see the [Access logging API]({{% versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/options/als/als.proto.sk/" %}}). 

### Match conditions on validation webhook

> You can now specify match condititions on the Gloo Gateway or Kubernetes validating admission webhook level to filter the resources that you want to include or exclude from validation. Match conditions are written in [CEL (Common Expression Language)](https://cel.dev) language. 

Reviewer: ```suggestion
You can now specify match conditions on the Gloo Gateway or Kubernetes validating admission webhook level to filter the resources that you want to include or exclude from validation. Match conditions are written in [CEL (Common Expression Language)](https://cel.dev). 
```

### Breaking changes

> **Envoy version 1.32/ 1.33 upgrade**

Author: new content

> **Envoy version 1.32/ 1.33 upgrade**

Reviewer: ```suggestion
Does this mean 1.32 to 1.33? or if not, still would put a space before the slash
**Envoy version 1.32 to 1.33 upgrade**
```

> The Envoy dependency in Gloo Gateway 1.19 was upgraded from 1.31.x to 1.33.x. This change includes the following breaking changes. For more information about these changes, see the [Envoy changelog documentation](https://www.envoyproxy.io/docs/envoy/latest/version_history/v1.33/v1.33).

Reviewer: maybe add these are upstream breaking changes?
```suggestion
The Envoy dependency in Gloo Gateway 1.19 was upgraded from 1.31.x to 1.33.x. This change includes the following upstream breaking changes. For more information about these changes, see the [Envoy changelog documentation](https://www.envoyproxy.io/docs/envoy/latest/version_history/v1.33/v1.33).
```

* **Trust internal addresses**: By default Envoy is configured to trust internal addresses. However, in an upcoming release of Envoy, this behavior is changed and internal addresses must be added to the `internal_address_config` for Envoy to trust them. For example, if you have tooling, such as probes on your private network, make sure to include these IP addresses or CIDR ranges in the `internal_address_config` field. To try out or enable this upcoming behavior change, you can set the `envoy.reloadable_features.explicit_internal_address_config` runtime guard to `true`. For more information, see the related [pull request](https://github.com/envoyproxy/envoy/pull/36221/files) in Envoy.
* **Access log handlers**: Access log handlers that are added by filters are now evaluated before access log handlers that are configured in the `access_log` configuration. To disable this behavior, you can set the `envoy.reloadable_features.filter_access_loggers_first` runtime guard flag to `false`.

> * **Cluster name change in Kuberetes Gateway API**: When using the Kubernetes Gateway API alongside Gloo Edge APIs and you route to Kubernetes Services or Upstreams, the Envoy cluster name format is chnaged to extract more about the service. The new format uses underscores to list service details, such as `upstreamName_upstreamNs_svcNs_svcName_svcPort`. If you enabled the Kubernetes Gateway API integration with `kubeGateway.enabled=true`, both the Gloo Edge and Kubernetes Gateway API proxies use the same format for these clusters. 

Reviewer: ```suggestion
* **Cluster name change in Kuberetes Gateway API**: When using the Kubernetes Gateway API alongside Gloo Edge APIs and you route to Kubernetes Services or Upstreams, the Envoy cluster name format is changed to extract more details about the service. The new format uses underscores to list service details, such as `upstreamName_upstreamNs_svcNs_svcName_svcPort`. If you enabled the Kubernetes Gateway API integration with `kubeGateway.enabled=true`, both the Gloo Edge and Kubernetes Gateway API proxies use the same format for these cluster names. 
```
